<launch>
    <arg name="sim_robot" default="true"/>
    <arg name="robot_ip" unless="$(arg sim_robot)"/>
    <arg name="sim_sensor" default="true"/>

    <arg name="rviz" default="true"/>
    <arg name="box_side" default="0.2"/>
    <arg name="box_x" default="0.15"/>
    <arg name="box_y" default="0.4"/>
    <arg name="box_parent_link" default="workcell_base"/>

    <!-- Bringup the robot -->
    <include file="$(find pick_and_place)/launch/bringup_robot.launch" />


    <!-- Publish the location of the camera -->
    <node pkg="tf" type="static_transform_publisher" name="camera_tf_broadcaster"
       args="0.75 0.5 .75 -2.012 0.829 -1.571 base_link camera_link 10" />

    <!-- Launch simulated sensor -->
    <group if="$(arg sim_sensor)">
        <!-- Rotate to suit simulation -->
       <node pkg="tf" type="static_transform_publisher" name="camera_tf_remap"
           args="0 0 0 0 1.57079 0 camera_link sim_camera_link 1" />
       <node name="sensor_sim" pkg="pick_and_place" type="sensor_simulator_3d" output="screen"/>
    </group>

    <!-- Launch real sensor -->
    <group unless="$(arg sim_sensor)">
         <!-- Launch camera node -->
         <include file="$(find realsense2_camera)/launch/rs_rgbd.launch" />
    </group>

    <!-- Launch vision node -->
    <include file="$(find pick_and_place_perception)/launch/processing_node.launch" />

    <!-- Launch planner node -->
    <node name="pick_and_place_node" pkg="pick_and_place" type="pick_and_place_node" output="screen"/>

    <!-- Launch visualization -->
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="pick_and_place_viz"
        args="-d $(find pick_and_place)/config/pick_and_place.rviz" />

</launch>

